sources/api/static-pods/src/static_pods.rs

```diff
@@ -54,13 +55,27 @@ where
     S2: AsRef<[u8]>,
 {
     let name = name.as_ref();
+    let manifest = manifest.as_ref();
 
     let dir = Path::new(STATIC_POD_DIR);
     fs::create_dir_all(&dir).context(error::Mkdir { dir: &dir })?;
-    let path = dir.join(name);
-    // Create the file if it does not exist, completely replace its contents
-    // if it does (constitutes an update).
-    fs::write(path, manifest).context(error::ManifestWrite { name })?;
+
+    // Create the pod manifest file as a temporary file first and finish writing to it
+    // before swapping any files out
+    let mut temp_manifest_file = NamedTempFile::new_in(dir).context(error::CreateTempfile)?;
+    temp_manifest_file
+        .write(manifest)
+        .context(error::ManifestWrite { name })?;
```

Comment:
This fixes the case where a manifest exists that kubelet is watching, but it seems like it has the same potential problem for a newly created manifest: kubelet could try to load it before we've written any content, or before we've finished writing it.

Reply:
It loads any file in that directory regardless of name?  I was assuming it'd look for at least _some_ kind of pattern.  If that's the case, we could put the temp file in a parent directory, or a new temp directory we create next to the target directory..?